internal/keycloakb/accreditationsmodule.go

```diff
@@ -0,0 +1,150 @@
+	Type       *string `json:"type,omitempty"`
+	ExpiryDate *string `json:"expiryDate,omitempty"`
+}
+
+// NewAccreditationsModule creates an accreditations module
+func NewAccreditationsModule(keycloakClient AccredsKeycloakClient, confDBModule AdminConfigurationDBModule, logger Logger) AccreditationsModule {
+	return &accredsModule{
+		keycloakClient: keycloakClient,
+		confDBModule:   confDBModule,
+		logger:         logger,
+	}
+}
+
+func (am *accredsModule) GetUserAndPrepareAccreditations(ctx context.Context, accessToken, realmName, userID, condition string) (kc.UserRepresentation, int, error) {
+	var kcUser kc.UserRepresentation
+
+	// Gets the realm
+	var realm, err = am.keycloakClient.GetRealm(accessToken, realmName)
+	if err != nil {
+		am.logger.Warn(ctx, "msg", "getKeycloakRealm: can't get realm from KC", "err", err.Error())
+		return kcUser, 0, errorhandler.CreateInternalServerError("keycloak")
+	}
+
+	// Retrieve admin configuration from configuration DB
+	var rac configuration.RealmAdminConfiguration
+	rac, err = am.confDBModule.GetAdminConfiguration(ctx, *realm.Id)
+	if err != nil {
+		am.logger.Warn(ctx, "msg", "CreateAccreditations: can't get admin configuration", "err", err.Error())
+		return kcUser, 0, errorhandler.CreateInternalServerError("keycloak")
+	}
+
+	// Evaluate accreditations to be created
+	var newAccreds []string
+	for _, modelAccred := range rac.Accreditations {
+		if modelAccred.Condition == nil || *modelAccred.Condition == condition {
+			var expiry *string
+			expiry, err = am.convertDurationToDate(ctx, *modelAccred.Validity)
+			if err != nil {
+				return kcUser, 0, err
+			}
+			var newAccreditationJSON, _ = json.Marshal(AccreditationRepresentation{
+				Type:       modelAccred.Type,
+				ExpiryDate: expiry,
+			})
+			newAccreds = append(newAccreds, string(newAccreditationJSON))
+		}
+	}
+
+	// Get the user from Keycloak
+	kcUser, err = am.keycloakClient.GetUser(accessToken, realmName, userID)
+	if err != nil {
+		am.logger.Warn(ctx, "msg", "CreateAccreditations: can't get Keycloak user", "err", err.Error(), "realm", realmName, "user", userID)
+		return kcUser, 0, err
+	}
+
+	// Update attributes in kcUser
+	// If no new accreditation, return
+	var added = 0
+	if len(newAccreds) > 0 {
+		var kcAccreds = kcUser.GetAttribute(constants.AttrbAccreditations)
+		for _, newAccred := range newAccreds {
+			if !validation.IsStringInSlice(kcAccreds, newAccred) {
+				kcAccreds = append(kcAccreds, newAccred)
+				added++
+			}
+		}
+		kcUser.SetAttribute(constants.AttrbAccreditations, kcAccreds)
+	}
+
+	return kcUser, added, nil
+}
+
+func (am *accredsModule) CreateAccreditations(ctx context.Context, accessToken, realmName, userID, condition string) error {
+	var kcUser, count, err = am.GetUserAndPrepareAccreditations(ctx, accessToken, realmName, userID, condition)
+	if err != nil || count == 0 {
+		return err
+	}
+
+	// Update user in Keycloak
+	err = am.keycloakClient.UpdateUser(accessToken, realmName, userID, kcUser)
```

Comment:
Is it really the matter of this module to update the user ?
Maybe this accreditations component could only be responsible to create the array of accreditations to persists into the user and let the upper layer (component) to update the user